From da27a39133568c732cb00820c5a7d94a4abc1c69 Mon Sep 17 00:00:00 2001
From: Liu Ying <b17645@freescale.com>
Date: Wed, 18 Feb 2009 11:26:53 +0800
Subject: [PATCH] ENGR00107856 IPUv3: Add RGB565 and LVDS666 pixel formats support for DC mapping

Add RGB565 and LVDS666 pixel formats support for DC mapping.

Signed-off-by: Liu Ying <b17645@freescale.com>
---
 drivers/mxc/ipu3/ipu_common.c |    2 ++
 drivers/mxc/ipu3/ipu_disp.c   |   16 ++++++++++++++++
 include/linux/ipu.h           |    2 ++
 3 files changed, 20 insertions(+), 0 deletions(-)

diff --git a/drivers/mxc/ipu3/ipu_common.c b/drivers/mxc/ipu3/ipu_common.c
index a6e5c99..0ee40e8 100644
--- a/drivers/mxc/ipu3/ipu_common.c
+++ b/drivers/mxc/ipu3/ipu_common.c
@@ -1720,6 +1720,8 @@ ipu_color_space_t format_to_colorspace(uint32_t fmt)
 	case IPU_PIX_FMT_RGB32:
 	case IPU_PIX_FMT_RGBA32:
 	case IPU_PIX_FMT_ABGR32:
+	case IPU_PIX_FMT_LVDS666:
+	case IPU_PIX_FMT_LVDS888:
 		return RGB;
 		break;
 
diff --git a/drivers/mxc/ipu3/ipu_disp.c b/drivers/mxc/ipu3/ipu_disp.c
index 0670018..d6aced7 100644
--- a/drivers/mxc/ipu3/ipu_disp.c
+++ b/drivers/mxc/ipu3/ipu_disp.c
@@ -565,6 +565,18 @@ void _ipu_init_dc_mappings(void)
 	_ipu_dc_map_config(2, 0, 15, 0xFF);
 	_ipu_dc_map_config(2, 1, 23, 0xFF);
 	_ipu_dc_map_config(2, 2, 7, 0xFF);
+
+	/* IPU_PIX_FMT_RGB565 */
+	_ipu_dc_map_clear(3);
+	_ipu_dc_map_config(3, 0, 4, 0xF8);
+	_ipu_dc_map_config(3, 1, 10, 0xFC);
+	_ipu_dc_map_config(3, 2, 15, 0xF8);
+
+	/* IPU_PIX_FMT_LVDS666 */
+	_ipu_dc_map_clear(4);
+	_ipu_dc_map_config(4, 0, 5, 0xFC);
+	_ipu_dc_map_config(4, 1, 13, 0xFC);
+	_ipu_dc_map_config(4, 2, 21, 0xFC);
 }
 
 int _ipu_pixfmt_to_map(uint32_t fmt)
@@ -577,6 +589,10 @@ int _ipu_pixfmt_to_map(uint32_t fmt)
 		return 1;
 	case IPU_PIX_FMT_YUV444:
 		return 2;
+	case IPU_PIX_FMT_RGB565:
+		return 3;
+	case IPU_PIX_FMT_LVDS666:
+		return 4;
 	}
 
 	return -1;
diff --git a/include/linux/ipu.h b/include/linux/ipu.h
index a28d44e..34f5fcd 100644
--- a/include/linux/ipu.h
+++ b/include/linux/ipu.h
@@ -87,6 +87,8 @@ typedef enum {
 /*! @{ */
 #define IPU_PIX_FMT_GENERIC fourcc('I', 'P', 'U', '0')	/*!< IPU Generic Data */
 #define IPU_PIX_FMT_GENERIC_32 fourcc('I', 'P', 'U', '1')	/*!< IPU Generic Data */
+#define IPU_PIX_FMT_LVDS666 fourcc('L', 'V', 'D', '6')	/*!< IPU Generic Data */
+#define IPU_PIX_FMT_LVDS888 fourcc('L', 'V', 'D', '8')	/*!< IPU Generic Data */
 /*! @} */
 /*! @name RGB Formats */
 /*! @{ */
-- 
1.5.4.4

